crates/matrix-sdk-crypto/src/session_manager/group_sessions.rs

```diff
@@ -283,14 +284,21 @@ impl GroupSessionManager {
         let mut share_infos = BTreeMap::new();
         let mut withheld_devices = Vec::new();
 
-        let encrypt = |device: Device, session: OutboundGroupSession| async move {
-            let encryption_result = device.maybe_encrypt_room_key(session).await?;
+        // XXX is there a way to do this that doesn't involve cloning the
```

Comment:
I don't think there is, you are moving the `Device` into a spawned task. The task itself might be moved between threads.

Reply:
my thinking went: we have an `Arc<CryptoStoreWrapper>` which is passed into this method, and which we know will live as long as any of those tasks (because the tasks are all spawned and `join_all`-ed inside this method). So shouldn't I be able to borrow a reference to the `CryptoStoreWrapper` contained in that Arc, and just pass a copy of that reference into each spawned task?